src/components/controls/Checkbox.tsx

```diff
@@ -4,31 +4,33 @@ import * as s from './checkbox.scss';
 interface ICheckboxProps {
     checked: boolean;
     text: string;
-    onClick(event: any): void;
+    onClick(): void;
 }
 
 class Checkbox extends React.Component<ICheckboxProps, {}> {
-    private doNothing() {
-        // Empty
+    private onClick = (event: React.ChangeEvent<HTMLElement>) => {
+        this.props.onClick();
+        event.stopPropagation();
+        event.preventDefault();
     }
 
-    public render(): any {
-        const onClick = (event: React.MouseEvent<HTMLElement>) => {
-            this.props.onClick(event);
-            event.stopPropagation();
-            event.preventDefault();
-        };
+    private onChange = (event: React.MouseEvent<HTMLElement>) => {
+        this.props.onClick();
+        event.stopPropagation();
+        event.preventDefault();
+    }
 
+    public render() {
         return (
             <label
-                onClick={onClick}
+                onClick={this.onChange}
                 className={s.container}
             >
                 {this.props.text}
                 <input
-                    type={'checkbox'}
+                    type='checkbox'
                     checked={this.props.checked}
-                    onChange={this.doNothing}
+                    onChange={this.onClick}
```

Comment:
Why were these changes made? I don't see any checkboxes in the scope of this PR (linkView)

Reply:
Checkbox was used in linkview, but it was now moved to reitinsuunta list. I noticed the component had many flaws, so I improved it a bit. onClickHandlers was created in the render function, onClick required event parameter, render function had specified it returns any, curly braces around string in JSX.

Reply:
I however reverted the onChange function, since that seems to not be working as I planned it would